### ISSUE TITLE
Keyboard navigation doesn't work when selected cell is out of viewport.

### ISSUE THREAD

**Issue by zindel** [mleibman/SlickGrid#143]
Hi,

first of all let me thank you for all your hard work on slickgrid. This useful piece of software made my life a lot easier.

I noticed this issue with the keyboard navigation (we use row-based navigation in our project, so when I refer to it I mean Up/Down keys functioning). Steps to reproduce:
1. Click on cell to select it.
2. Scroll grid down enough for the selected cell to disappear from viewport.
3. Press Down.
4. Nothing happens. I think the correct behavior would be to automatically do 'scrollIntoView' for the selected cell and then trigger the 'Down' key handler to move selection.

I've created a 'hackish' fix for our own project, although I'm not sure how correct is it to be applied for the whole scope of use cases. The main idea of it was to save not only selected row in the rowCache but selected+1 & selected-1 rows as well. It seems to work more or less fine, but sometimes I'm noticing weird bugs (likely connected with my patch).  Let me know please if you have an idea how it could be properly fixed (even you don't have time/wish to fix it yourself:)) I may try to implement and send patch for your review.

Thank you.
